introduction/modules/modules.md

```diff
@@ -15,41 +18,51 @@ Connectors are Valtimo configurable components that can be used to connect to ex
 framework to create, configure, and use connectors. Certain Valtimo modules include their own connectors that become
 available automatically for configuration when starting a Valtimo instance.
 
+> **_NOTE:_** Connectors will be replaced by Plugins in the future
+
 ### Contract
 
-Contract contains certain interfaces and events that Valtimo uses in several other modules. Apart from this, contract
-sanitizes endpoint output when exceptions occur. A liquibase runner is available that allows other modules to define
-liquibase configuration that is run on startup. Finally, Contract includes several helper classes, like the
-`QueryDialectHelper` bean that can be used in criteria queries to add JSON column clauses to the query.
+Contract contains certain interfaces and events that Valtimo uses in several other modules. Consider it the 'glue'
+between a module and Core.
+
+Apart from this, contract sanitizes endpoint output when exceptions occur. A liquibase runner is available that allows
+other modules to define liquibase configuration that is run on startup. Finally, Contract includes several helper
+classes, like the `QueryDialectHelper` bean that can be used in criteria queries to add JSON column clauses to the
+query.
 
 ### Core
 
-Core contains and builds upon the Camunda process engine that enables Valtimo to run processes. Several endpoints and
-functions are available that provide task data and process data. It also includes choice fields. A generic way to create
-lists of values, that can be used by the application. Some security features used to control access to any Valtimo
-endpoints are also part of this module.
+Core is the main module of Valtimo. It contains and builds upon the Camunda process engine that enables Valtimo to run
+processes. Several endpoints and functions are available that provide task data and process data.
+
+It also includes choice fields. A generic way to create lists of values, that can be used by the application. Some
+security features used to control access to any Valtimo endpoints are also part of this module.
 
 ### Documents
 
-A document is the starting point of the case and the main data storage for that case. One or more processes can be
-linked to a case, which allows for the document to be created when the case is started. The content of the document is 
-defined in a document definition, in JSON schema. Users interact with the document via user tasks; document data can be 
-loaded into task forms, and user input can be stored in the document after completing a task.
+A document is the main data storage entity that can contain data for one or more processes. It can be linked to a case,
```

Comment:
This is not completely correct. technically a document is the same as a case right now so it cannot be 'linked' to a case. 

```suggestion
A document is the main data storage entity that contains data of a case. Processes can act upon and change the data that is contained in the document. 
```